core/husarnet.cmake

```diff
@@ -81,16 +81,22 @@ endif()
 
 # Top level project files
 
-# So includes like "husarnet/something.h" do work too
-include_directories(${CMAKE_CURRENT_LIST_DIR}/..)
 include_directories(${CMAKE_CURRENT_LIST_DIR})
 
 file(GLOB core_SRC "${CMAKE_CURRENT_LIST_DIR}/*.cpp")
 list(APPEND husarnet_core_SRC ${core_SRC})
 
+# "Aliasing" (copying) core folder as "husarnet"
+# So includes like "husarnet/something.h" do work too
+set(TEMP_INCLUDE_DIR ${CMAKE_BINARY_DIR}/tempIncludes)
+file(MAKE_DIRECTORY ${TEMP_INCLUDE_DIR})
+file(COPY ../../core/ DESTINATION ${TEMP_INCLUDE_DIR}/husarnet/ FILES_MATCHING PATTERN "*.h")
+
 # Join all of the above
 add_library(husarnet_core STATIC ${husarnet_core_SRC})
 
+target_include_directories(husarnet_core PUBLIC ${TEMP_INCLUDE_DIR})
```

Comment:
Kinda feels like that line belongs to that section above a little bit more

Reply:
I agree, but as per CMake docs: 
target_include_directories
Specifies include directories to use when compiling a given target. The named <target> must have been created by a command such as add_library (...)

implies `add_library` must come before any `target_include_directories` and I can confirm that this indeed does not work after moving up